test/e2e/util/reqserving/verifypods.go

```diff
@@ -0,0 +1,135 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reqserving
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	supportutil "github.com/openshift/hypershift/support/util"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// VerifyRequestServingPodDistribution verifies that only the correct request serving pods
+// are scheduled on request serving nodes with expected replica counts and proper labeling.
+func VerifyRequestServingPodDistribution(ctx context.Context, hc *hyperv1.HostedCluster) error {
+	cpNamespace := fmt.Sprintf("%s-%s", hc.Namespace, hc.Name)
+	client, err := e2eutil.GetClient()
+	if err != nil {
+		return fmt.Errorf("failed to get cluster client: %w", err)
+	}
+
+	var lastErr error
+	pollCtx, cancel := context.WithTimeout(ctx, 15*time.Minute)
+	defer cancel()
+	err = wait.PollUntilContextCancel(pollCtx, 30*time.Second, true, func(pctx context.Context) (bool, error) {
+		var errs []error
+
+		// Build set of request-serving nodes for this HostedCluster
+		requestServingNodes := &corev1.NodeList{}
+		expectedClusterIdentifier := fmt.Sprintf("%s-%s", hc.Namespace, hc.Name)
+		if err := client.List(pctx, requestServingNodes, crclient.MatchingLabels{
+			hyperv1.RequestServingComponentLabel: "true",
+			hyperv1.HostedClusterLabel:           expectedClusterIdentifier,
+		}); err != nil {
+			lastErr = fmt.Errorf("failed to list request serving nodes: %w", err)
+			return false, nil
+		}
+		if len(requestServingNodes.Items) == 0 {
+			lastErr = fmt.Errorf("no request serving nodes found for HostedCluster %s", hc.Name)
+			return false, nil
+		}
+		requestServingNodeNames := make(map[string]bool)
+		for _, node := range requestServingNodes.Items {
+			requestServingNodeNames[node.Name] = true
+		}
+
+		// Verify pods are on request-serving nodes and properly labeled
+		requestServingPods := &corev1.PodList{}
+		if err := client.List(pctx, requestServingPods,
+			crclient.InNamespace(cpNamespace),
+			crclient.MatchingLabels{hyperv1.RequestServingComponentLabel: "true"}); err != nil {
+			lastErr = fmt.Errorf("failed to list request serving pods: %w", err)
+			return false, nil
+		}
+
+		for _, pod := range requestServingPods.Items {
+			if pod.Spec.NodeName == "" {
+				errs = append(errs, fmt.Errorf("request serving pod %s is not scheduled to any node", pod.Name))
+				continue
+			}
+			if !requestServingNodeNames[pod.Spec.NodeName] {
+				errs = append(errs, fmt.Errorf("request serving pod %s is scheduled on non-request-serving node %s", pod.Name, pod.Spec.NodeName))
```

Comment:
These checks look exactly like those in in create_cluster_test.go here: https://github.com/openshift/hypershift/blob/main/test/e2e/create_cluster_test.go#L2156-L2157
Does it make sense to remove the original test from create_cluster_test.go and have this covered here? Otherwise we will have some unnecessary overlap. Thanks